From 7d1fba3418643b8d4b3c74157c59cd090aaf4680 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:10 +0800
Subject: [PATCH] arm64/mpam: alloc/mon capable/enabled debug
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 7d1fba3418643b8d4b3c74157c59cd090aaf4680


hulk inclusion
category: feature
feature: ARM MPAM support
bugzilla: 48265
CVE: NA

--------------------------------

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/kernel/mpam.c b/arch/arm64/kernel/mpam.c
index 6383c7d60135..d14693e139bb 100644
--- a/arch/arm64/kernel/mpam.c
+++ b/arch/arm64/kernel/mpam.c
@@ -1221,6 +1221,14 @@ static void mpam_domains_init(struct resctrl_resource *r)
 			rr->num_mon = MPAMF_IDR_NUM_MON(val);
 		}
 
+		pr_info("%s: type: %s; alloc = %d %d; mon = %d %d\n",
+			__func__, types[n->type],
+			r->alloc_capable,
+			r->alloc_enabled,
+			r->mon_capable,
+			r->mon_enabled
+			);
+
 		d->cpus_list = n->cpus_list;
 
 		d->ctrl_val = kmalloc_array(rr->num_partid, sizeof(*d->ctrl_val), GFP_KERNEL);
-- 
2.26.2

